### PR TITLE
Replace Mesos and Zookeeper fixtures with USI.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This unifies our Mesos and Zookeeper test setup across the Jenkins
plugin, Metronome, Marathon and USI.

Blocked by mesosphere/usi#108

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -149,6 +150,7 @@ object Dependency {
     val Mockito = "1.10.19"
     val ScalaCheck = "1.13.5"
     val ScalaTest = "3.0.4"
+    val UsiTestUtil = "0.1.4-a97fc25-SNAPSHOT"
```
Comment (timcharper) [inline project/Dependencies.scala]: did we not merge yet? I would like to add a rule for snapshot packages in USI to expire after a period (30 days?), we should not avoid using them in PRs and just use them for testing changes and coding

Comment (jeschkies) [reply project/Dependencies.scala]: > we should not avoid using them in PRs and just use them for testing changes and coding

The is no USI master package with the required changes yet. I'm not sure I understand your comment. The snapshot dependency is not supposed to land on Marathon master.

**Comment by jeschkies** [comment]
Somewhere in `LeaderIntegrationTest` we spent more than 2 1/5 minutes between lines `274` and `305`. This causes the test to timeout. Checkout the log

```
    158 WARN [10:01:32 BackupRestoreIntegrationTest-LocalMarathon-32906] I1009 10:01:32.257115 15183 sched.cpp:363] No credentials provided. Attempting to register without authentication
    159 WARN [10:01:32 BackupRestoreIntegrationTest-LocalMarathon-32906] I1009 10:01:32.259855 15179 sched.cpp:751] Framework registered with 18ec44db-47b8-44a2-9d3a-8464c9d4a7ce-0000
    160 INFO [10:04:01 BackupRestoreIntegrationTest] SSEStream: Connecting
    161 INFO [10:04:01 BackupRestoreIntegrationTest] SSEStream: Acquiring connection to http://localhost:32906
```

Marathon with port `32906` is up at 10:01:32 but we did not connect to the SSE stream until 10:04:01.

**Comment by jeschkies** [comment]
The time is spent waiting to reconnect. See line `886` in `MarathonTest`:
```scala
scheduler.scheduleOnce(patienceConfig.timeout / 2) { iter() }
```

File: tests/integration/src/test/scala/mesosphere/marathon/integration/LeaderIntegrationTest.scala [hunk 2]
```diff
@@ -245,8 +245,6 @@ class BackupRestoreIntegrationTest extends LeaderIntegrationTest {
 
       val tmpBackupFile = File.createTempFile("marathon", "BackupRestoreIntegrationTest")
 
-      waitForSSEConnect()
```
Comment (timcharper) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/LeaderIntegrationTest.scala]: Hmm, why did you remove this line? These were put in to prevent flakes related to missed events

Comment (timcharper) [reply tests/integration/src/test/scala/mesosphere/marathon/integration/LeaderIntegrationTest.scala]: Is it because the test support code already waited for an SSE connection event?

File: project/Dependencies.scala (hunk 1 above)
Comment (timcharper) [reply project/Dependencies.scala]: Ok, should we mark this as a WIP PR then? We do use snapshot packages for precisely the reason you outline

**Comment by timcharper** [comment]
> Locally I see
> 
> ```
> WARN [16:25:51 PoolGateway] [2 (WaitingForEndOfResponseEntity)] Ongoing request [GET /v2/events Empty] was dropped because pool is shutting down
> ```
> 
> And thus no deployment event is received. Any ideas, @timcharper?

I think this is being emitted when the Akka actor system is being shut down.

File: project/Dependencies.scala (hunk 1 above)
Comment (jeschkies) [reply project/Dependencies.scala]: The dependency is already `0.1.7`. I'm not sure what you mean.
